src/lib/cloud-sdk-api.js

```diff
@@ -411,15 +411,14 @@ class CloudSdkAPI {
   }
 
   async _waitForEnv(namespace, ...allowedStates) {
-    return await this._waitForJson(
-      (releaseState) => allowedStates.includes(releaseState),
-      async () => {
-        const status = await this._devConsoleClient.doGet(
-          `/api/releases/${namespace}/status`
-        );
-        return status.releases?.status[this._cmReleaseId]?.releaseState;
-      }
-    );
+    return this._waitForJson(
+      (releaseState) =>
+        allowedStates.includes(
+          releaseState.releases?.status[this._cmReleaseId]?.releaseState
+        ),
+      async () =>
+        await this._devConsoleClient.doGet(`/api/releases/${namespace}/status`)
+    ).then((json) => json.releases.status[this._cmReleaseId].releaseState);
```

Comment:
I don't like that we are mixing `async/await` with `Promise.then()` styles. Also the request can be simplified by removing the async and await. I think it already returns a promise.

```suggestion
    let getReleaseState = (json) => json.releases?.status[this._cmReleaseId]?.releaseState;
    let json = await this._waitForJson(
      (json) => allowedStates.includes(getReleaseState(json)),
      this._devConsoleClient.doGet(`/api/releases/${namespace}/status`));
    return getReleaseState(json);
```

BTW, are we sure that  the final `json` always contains the expected structure?

Reply:
Resolved: [9999c65](https://github.com/adobe/aio-cli-plugin-aem-rde/pull/49/commits/9999c65727d7274f8114ca5bdba3b4131cb0ea65)